kimchi/src/verifier.rs

```diff
@@ -471,27 +471,19 @@ where
             proof.prev_challenges.len(),
         ));
     }
+    if proof.public.len() != index.public {
+        return Err(VerifyError::IncorrectPubicInputLength(index.public));
+    }
```

Comment:
oh right I see it now, we're missing a spec comment here but it's unrelated to this PR